vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,364 @@
+		return err
+	}
+	err = d.Set("telephone", orgUser.User.Telephone)
+	if err != nil {
+		return err
+	}
+	err = d.Set("instant_messaging", orgUser.User.IM)
+	if err != nil {
+		return err
+	}
+	err = d.Set("is_enabled", orgUser.User.IsEnabled)
+	if err != nil {
+		return err
+	}
+	err = d.Set("is_locked", orgUser.User.IsLocked)
+	if err != nil {
+		return err
+	}
+	err = d.Set("deployed_vm_quota", orgUser.User.DeployedVmQuota)
+	if err != nil {
+		return err
+	}
+	err = d.Set("stored_vm_quota", orgUser.User.StoredVmQuota)
+	if err != nil {
+		return err
+	}
+	err = d.Set("role", orgUser.User.Role.Name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates an OrgUser from data provided in the resource
+func resourceVcdOrgUserCreate(d *schema.ResourceData, meta interface{}) error {
+
+	userData, adminOrg, err := resourceToUserData(d, meta)
+	if err != nil {
+		return err
+	}
+	if userData.Password == "" {
+		return fmt.Errorf(`no password provided with either "password"" or "password_file" properties`)
+	}
+	_, err = adminOrg.CreateUserSimple(*userData)
+	if err != nil {
+		return err
+	}
+	return resourceVcdOrgUserRead(d, meta)
+}
+
+// Deletes an OrgUser
+func resourceVcdOrgUserDelete(d *schema.ResourceData, meta interface{}) error {
+
+	takeOwnership := d.Get("take_ownership").(bool)
+	orgUser, _, err := resourceToOrgUser(d, meta)
+	if err != nil {
+		return err
+	}
+	return orgUser.Delete(takeOwnership)
+}
+
+// Reads the OrgUser from vCD and fills the resource container appropriately
+func resourceVcdOrgUserRead(d *schema.ResourceData, meta interface{}) error {
+
+	orgUser, adminOrg, err := resourceToOrgUser(d, meta)
+	if err != nil {
+		return err
+	}
+	return setOrgUserData(d, orgUser, adminOrg)
+}
+
+// Updates an OrgUser with the data passed through the resource
+func resourceVcdOrgUserUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	orgUser, _, err := resourceToOrgUser(d, meta)
+	if err != nil {
+		return err
+	}
+	userData, _, err := resourceToUserData(d, meta)
+	if err != nil {
+		return err
+	}
+	err = orgUser.UpdateSimple(*userData)
+	if err != nil {
+		return err
+	}
+	return resourceVcdOrgUserRead(d, meta)
+}
+
+// Imports an OrgUser into Terraform state
+// This function task is to get the data from vCD and fill the resource data container
+func resourceVcdOrgUserImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 2 {
+		return nil, fmt.Errorf("resource name must be specified as org.org_user")
+	}
+	orgName, userName := resourceURI[0], resourceURI[1]
+
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrg(orgName)
+	if err != nil || adminOrg == (govcd.AdminOrg{}) {
+		return nil, fmt.Errorf(errorRetrievingOrg, orgName)
+	}
+
+	user, err := adminOrg.FetchUserByName(userName, false)
+	if err != nil {
+		return nil, govcd.ErrorEntityNotFound
+	}
+
+	err = setOrgUserData(d, user, &adminOrg)
```

Comment:
@Didainius can comment more, but according his imports it should set only a few details and user will fetch all latter details with `Refresh`

Reply:
Yeah. It's probably not necessary if`read` sets all fields as import automatically triggers `refresh` however this probably can't make it worse.

Here's example of `import` triggering `read` 
```bash
$  terraform import vcd_lb_app_rule.test3 my-org.my-org-vdc.my-edge-gw.script2
vcd_lb_app_rule.test3: Importing from ID "my-org.my-org-vdc.my-edge-gw.script2"...
vcd_lb_app_rule.test3: Import complete!
  Imported vcd_lb_app_rule
vcd_lb_app_rule.test3: Refreshing state... [id=applicationRule-42]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

Reply:
There is no user refresh

Reply:
Probably both ways are fine. `refresh` is being triggered automatically so if `id` is set `refresh` would fill in all the other fields anyway.